test/e2e/e2e.go

```diff
@@ -694,18 +699,17 @@ var _ = ginkgo.Describe("e2e egress IP validation", func() {
 			targetNodeLogs = strings.TrimSuffix(targetNodeLogs, "\n")
 			logLines := strings.Split(targetNodeLogs, "\n")
 			lastLine := logLines[len(logLines)-1]
-			var found bool
-			for _, verifyIP := range verifyIPs {
-				if strings.Contains(lastLine, verifyIP) {
-					found = true
+			for i := 0; i < len(verifyIPs); i++ {
+				if strings.Contains(lastLine, verifyIPs[i]) {
+					verifyIPs = removeSliceElement(verifyIPs, i)
 					break
```

Comment:
if you find one verify IP you are going to break here, dont you want to validate you go through multiple egress nodes? I'm not really sure what verifyIPs are

Reply:
`verifyIPs` are the IPs we want to verify as having been seen on the external test container. We target the external test container from our test pods, collect its logs and verify that the logs have traces of the egress IP the pod is supposed to be assigned to. This patch now changes `targetExternalContainerAndTest` a bit, because we need to target the external test container multiple times until we verify that all IPs provided by `verifyIPs` have been verified. The way I am doing this is by passing it a slice of `verifyIPs` and removing each item when it's been found, the step only passes once `verifyIPs` is of length 0. I want to break here because `targetExternalContainerAndTest` initiates only a single connection at a time, sequentially: hence we perform one connection attempt, check that the IP seen is expected, remove it from the list of `verifyIPs`, see that it's length is not 0 and retry again. We do this until all IPs have been seen. If that never happens (because of a bug) the test fails.